processing/data_collection/gazette/spiders/pr_ponta_grossa.py

```diff
@@ -0,0 +1,56 @@
+from dateparser import parse
+import datetime as dt
+import re
+
+import scrapy
+
+from gazette.items import Gazette
+
+
+class PrPontaGrossaSpider(scrapy.Spider):
+    MUNICIPALITY_ID = '4119905'
+    name = 'pr_ponta_grossa'
+    allowed_domains = ['pontagrossa.pr.gov.br']
+    start_urls = ['http://www.pontagrossa.pr.gov.br/diario-oficial/']
+    ano_minimo = 2015
+
+    def parse(self, response):
+        """
+        @url http://www.pontagrossa.pr.gov.br/diario-oficial/
+        @returns requests 1
+        """
+        return self.scrape_page(response)
+
+    def scrape_page(self, response):
+        pdf_links = response.css(".view-content .field a")
+
+        pdf_infos = []
+        for pdf_link in pdf_links:
+            pdf_file_name = pdf_link.css("::attr(href)").extract_first()
+            pdf_link_text = pdf_link.css("::text").extract_first()
+            if "sem_atos" in pdf_file_name:
+                continue
+            pdf_link_info = re.search('.*/diario-oficial/_?(\d{4})-(\d{2})-(\d{2}).*.pdf', pdf_file_name)
+            ano = int(pdf_link_info.group(1))
+            if ano < self.ano_minimo:
+                continue
+            mes = pdf_link_info.group(2)
+            dia = pdf_link_info.group(3)
+            is_extra = "complementar" in pdf_link_text
+            pdf_infos.append({ "ano" : ano, "mes" : mes, "dia":dia, "url": pdf_file_name, "is_extra_edition": is_extra })
```

Comment:
Why not extracting this bit (lines 27-40) into a custom method? This method [could be a generator](https://cuducos.me/2017/08/11/geradores-do-python-o-que-sao-e-quais-problemas-resolvem.html) (way better than appending to a list).

This would like something like:

```python
    @staticmethod
    def pdfs_info(links):
        link_pattern = '.*/diario-oficial/_?(\d{4})-(\d{2})-(\d{2}).*.pdf'
        for link in links:
            file_name = pdf_link.css('::attr(href)').extract_first()
            link_text = pdf_link.css('::text').extract_first()

            if 'sem_atos' in file_name:
                continue

            info = re.search(link_pattern, file_name)
            year = int(info.group(1))
            if year < self.ano_minimo:
                continue

            month, day = int(info.group(2)), int(info.group(3))
            yield {
                'date': date(year, month, day),
                'is_extra_edition': 'complementar' in link_text
            }
```